src/daemon/absorber.cc

```diff
@@ -13,6 +13,17 @@ using namespace std::placeholders;
 namespace dist_clang {
 namespace daemon {
 
+namespace {
+
+bool ValidateClientVersion(ui32 client_version) {
```

Comment:
Looking at this function only I can say that this approach is not anticipated. The backward-compatible version should be the field in the Absorber config. Since the usual situation is when the clients (Emitters) are updated much more often than Absorbers, it's convenient to update this field without redistributing the new package.